examples/manipulation/behavior_cloning.py

```diff
@@ -0,0 +1,414 @@
+
+                # Compute action prediction loss
+                action_loss = F.mse_loss(pred_action, batch["actions"])
+
+                # Compute pose estimation loss (position + orientation)
+                pose_left_loss = self._compute_pose_loss(pred_left_pose, batch["object_poses"])
+                pose_right_loss = self._compute_pose_loss(pred_right_pose, batch["object_poses"])
+                pose_loss = pose_left_loss + pose_right_loss
+
+                # Combined loss with weights
+                total_loss = action_loss + pose_loss
+
+                # Backward pass
+                self._optimizer.zero_grad()
+                total_loss.backward()
+                self._optimizer.step()
+                torch.nn.utils.clip_grad_norm_(self._policy.parameters(), self._cfg["max_grad_norm"])
+
+                total_action_loss += action_loss
+                total_pose_loss += pose_loss
+                num_batches += 1
+
+            end_time = time.time()
+            backward_time = end_time - start_time
+
+            # Compute average losses
+            avg_action_loss = total_action_loss / num_batches if num_batches > 0 else 0.0
+            avg_pose_loss = total_pose_loss / num_batches if num_batches > 0 else 0.0
+
+            fps = (self._num_steps_per_env * self._env.num_envs) / (forward_time)
+            # Logging
+            if (it + 1) % 10 == 0:
+                current_lr = self._optimizer.param_groups[0]["lr"]
+
+                tf_writer.add_scalar("loss/action_loss", avg_action_loss, it)
+                tf_writer.add_scalar("loss/pose_loss", avg_pose_loss, it)
+                tf_writer.add_scalar("loss/total_loss", avg_action_loss + avg_pose_loss, it)
+                tf_writer.add_scalar("lr", current_lr, it)
+                tf_writer.add_scalar("buffer_size", self._buffer.size, it)
+                tf_writer.add_scalar("speed/forward", forward_time, it)
+                tf_writer.add_scalar("speed/backward", backward_time, it)
+                tf_writer.add_scalar("speed/fps", int(fps), it)
+
+                #
+                print("--------------------------------")
+                info_str = f" | Iteration:     {it + 1:04d}\n"
+                info_str += f" | Action Loss:   {avg_action_loss:.6f}\n"
+                info_str += f" | Pose Loss:     {avg_pose_loss:.6f}\n"
+                info_str += f" | Total Loss:    {avg_action_loss + avg_pose_loss:.6f}\n"
+                info_str += f" | Learning Rate: {current_lr:.6f}\n"
+                info_str += f" | Forward Time:  {forward_time:.2f}s\n"
+                info_str += f" | Backward Time: {backward_time:.2f}s\n"
+                info_str += f" | FPS:           {int(fps)}"
+                print(info_str)
+
+                if len(self._rewbuffer) > 0:
+                    tf_writer.add_scalar("reward/mean", np.mean(self._rewbuffer), it)
+
+            # Save checkpoints periodically
+            if (it + 1) % 50 == 0:
```

Comment:
Same. What about some checkpoint period parameter?

Reply:
same as above reply. removed the hardcoded number and defined `save_freq`